code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -511,4 +511,32 @@
 	ammo_type = /obj/item/ammo_casing/cap
 	matter = list(DEFAULT_WALL_MATERIAL = 600)
 	max_ammo = 7
-	multiple_sprites = 1
\ No newline at end of file
+	multiple_sprites = 1
+
+
+/obj/item/ammo_magazine/ma12gclip
+	name = "ammo clip (12g beanbag)"
+	icon_state = "12gclip_bean" //largely a codersprite, looks good enough. feel free to make a better one.
+	desc = "A metal clip for holding and quickly loading shotgun shells. This one is loaded with beanbag shells."
+	caliber = "12g"
+	ammo_type = /obj/item/ammo_casing/a12g/beanbag
+	matter = list(DEFAULT_WALL_MATERIAL = 1580) // beanbag shells x6 + 500 metal for the clip itself.
+	max_ammo = 6
+	multiple_sprites = 1
+	mag_type = SPEEDLOADER
```

Comment:
See previous comment, will adjust. 

I'm unsure of how to do the one-shell per click, as that was my original intent. As it stands it'll load every shell into the gun with a single click (which is once more not really what i wanted).